### PR TITLE
Site Editor: Fix Invisible Template Previews in the Sidebar

### PR DESCRIPTION
**Pull request by Copons**
## Description

A recent change on the FSE navigation sidebar's overflow accidentally hid the templates (and template parts) previews that were supposed to appear when hovering on the sidebar items.

This PR moves the preview SlotFill outside of the sidebar hidden overflow, and reposition it accordingly.

## How has this been tested?

- Open the Site Editor.
- Open the navigation sidebar.
- Enter either the Templates or Template Parts menu.
- Move the mouse over a template (or part).
- Make sure the template preview is visible, just below the header, and doesn't have z-index issues.

## Screenshot

Screenshot 2020-10-23 at 16 28 56

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/edit-site/src/components/navigation-sidebar/index.js [hunk 1]
```diff
@@ -18,8 +24,7 @@ const NavigationSidebar = () => {
 		<>
 			<NavigationToggle isOpen={ isNavigationOpen } />
 			<NavigationPanel isOpen={ isNavigationOpen } />
+			<NavigationPanelPreviewSlot />
```
Comment (vindl) [inline packages/edit-site/src/components/navigation-sidebar/index.js:27]: Cool, this seems like a better place for the slot, than nested within the panel. 👍

**Comment by mattwiebe** [comment]
Found an immediate z-index issue - might be hard to see, but it's the Post Title bleeding over. 
Screen Shot 2020-10-23 at 10 51 13

Still digging in but thought I'd put this in right away.

File: packages/interface/src/components/interface-skeleton/style.scss [hunk 2]
```diff
@@ -79,6 +79,10 @@ html.interface-interface-skeleton__html-container {
 
 }
 
+.interface-interface-skeleton__drawer {
+	position: relative;
```
Comment (vindl) [inline packages/interface/src/components/interface-skeleton/style.scss:83]: Why is this needed?

**Review by vindl — APPROVED** [review]
This looks good to me and tests well. 👍

I noticed something unrelated to this PR that we might want to follow up on: the content section is not displaying previews.

**Comment by Addison-Stavlo** [comment]
This is working well on my end as well.  I couldn't repro the z-index issue before, but it looks like the update should hopefully fix what @mattwiebe was noticing?

**Comment by mattwiebe** [comment]
It looks as though it's a conflict with the Cover block in my testing:

Screen Shot 2020-10-23 at 11 02 49

File: packages/interface/src/components/interface-skeleton/style.scss (hunk 2 above)
Comment (Copons) [reply packages/interface/src/components/interface-skeleton/style.scss]: Ah, it was needed because of the preview `left` value, but I've updated it now, and this positioning is not needed anymore.

**Comment by Copons** [comment]
@mattwiebe I can't repro the z-index issues as well, but I've raised the preview's from `1` to `30` (same as the interface skeleton header), which I hope is enough...
It's kinda hard to play around with z-index in the editor... 😕

**Comment by mattwiebe** [comment]
@Copons - can confirm that it's working as expected now, sits on top of Cover and everything in it.

Screen Shot 2020-10-23 at 11 10 55

**Review by mattwiebe — APPROVED** [review]

**Comment by vindl** [comment]
Nice catch @mattwiebe!
